### PR TITLE
Join relationships

### PR DESCRIPTION
**Pull request by prius2055**
### Join relationships

In this branch, I implemented the following;

- Created a vets table 
- Created a visits table with two foreign keys linking animals and vets
- Created a specializations table with two foreign keys linking species and vets
- Inserted given values following the project requirements
- Performed queries based on project instruction
- Updated the schema, data, and queries files.

Who was the last animal seen by William Tatcher?

join-table-result-1

How many different animals did Stephanie Mendez see?
join-table-query-result-2

List all vets and their specialties, including vets with no specialties.
join-table-query-result-3

List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
join-table-query-result-4

What animal has the most visits to vets?
join-table-query-result-5

Who was Maisy Smith's first visit?
join-table-query-result-6

Details for most recent visit: animal information, vet information, and date of visit.
join-table-query-result-7

How many visits were with a vet that did not specialize in that animal's species?
join-table-query-result-8

What specialty should Maisy Smith consider getting? Look for the species she gets the most.
join-table-query-result-9

### CODE REVIEW

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @prius2055, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
- You have created queries to answer all the questions correctly. You have used the correct table names, column names, join types, and WHERE clauses to filter and join the data. You have also used the correct ORDER BY, LIMIT, COUNT, DISTINCT, and GROUP BY clauses to sort, limit, aggregate, and group the data. You have followed the SQL syntax and formatting conventions. Your code will produce the expected results for each query. ✔️ 
- You have added screenshots of your queries results to the pull request description. This shows that you have run your queries with success and that your results are correct. This also helps to verify your work and demonstrate your skills. ✔️ 
- Good commit messages. :heavy_check_mark:
* No linter errors. :heavy_check_mark:
* Followed Github flow :heavy_check_mark:
* PR has a good title and summary :heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- For future pull requests, please notice that your branch name `relationships` does not follow the convention of being named after a feature or action. For example, you could name your branch `added-relationships`. This makes it easier to track the changes and the purpose of each branch.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql [hunk 1]
```diff
@@ -58,3 +58,140 @@ UPDATE animals
...
+
+
+/* Updating the visits table*/
+
+INSERT INTO visits(visit_dates,animals_id, vets_id)
+SELECT '2020-05-24',animals.id,vets.id
+FROM animals, vets
+WHERE animals.name = 'Agumon' AND vets.name = 'William Tatcher';
+                         
+INSERT INTO visits(visit_dates,animals_id, vets_id)
```
[the remark this answers is not in the document]
Comment (prius2055) [reply data.sql:100]: Fixed. Thank you

File: schema.sql [hunk 2]
```diff
@@ -40,3 +40,25 @@ ALTER TABLE animals ADD owner_id INTEGER;
...
+id SERIAL PRIMARY KEY,
+name TEXT,
+age INTEGER,
+date_of_graduation DATE
+);
+
+/* specializations table */
+CREATE TABLE specializations(
+species_id INTEGER, FOREIGN KEY(species_id) REFERENCES species(id),
+vets_id INTEGER, FOREIGN KEY(vets_id) REFERENCES vets(id)
```
[the remark this answers is not in the document]
Comment (prius2055) [reply schema.sql:56]: I have implemented this optional changes

File: schema.sql [hunk 3]
```diff
@@ -40,3 +40,25 @@ ALTER TABLE animals ADD owner_id INTEGER;
 ALTER TABLE animals
 ADD CONSTRAINT fk2 FOREIGN KEY (owner_id) REFERENCES owners (id);
 
+
+/* Create a table named vets */
+ 
+CREATE TABLE vets(
+id SERIAL PRIMARY KEY,
```
[the remark this answers is not in the document]
Comment (prius2055) [reply schema.sql]: This has been implemented. Thank you

[the remark this answers is not in the document]
Comment (topeogunleye) [reply schema.sql]: > This has been implemented. Thank you

@prius2055 Great Job 👍

**Review by franclobo — APPROVED** [review]
Hi @prius2055, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Correct Git flow ✔️ 
- Correct data ✔️ 
- Correct queries ✔️ 
- Correct schema ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**